scripts/pi-hole/php/header.php

```diff
@@ -337,34 +330,31 @@ function pidofFTL()
                         } else {
                             $pistatus = null;
                         }
-                        if ($pistatus === "1") {
+                        if ($pistatus === "53") {
                             echo '<span id="status"><i class="fa fa-circle text-green-light"></i> Active</span>';
                         } elseif ($pistatus === "0") {
                             echo '<span id="status"><i class="fa fa-circle text-red"></i> Offline</span>';
                         } elseif ($pistatus === "-1") {
                             echo '<span id="status"><i class="fa fa-circle text-red"></i> DNS service not running</span>';
+                        } elseif ($pistatus === "-2") {
+                            echo '<span id="status"><i class="fa fa-circle text-red"></i> Unknown</span>';
                         } else {
-                            echo '<span id="status"><i class="fa fa-circle text-orange"></i> Unknown</span>';
+                            echo '<span id="status"><i class="fa fa-circle text-orange"></i> DNS service on port '; echo $pistatus; echo '</span>';
```

Comment:
Please, replace this:

```php
echo '<span id="status"><i class="fa fa-circle text-orange"></i> DNS service on port '; echo $pistatus; echo '</span>';
```

with:
```php
echo '<span id="status"><i class="fa fa-circle text-orange"></i> DNS service on port ';
echo $pistatus;
echo '</span>';
```
or use concatenation.

Reply:
Also, if "$pistatus" is defined as **null** (on line 331), the output will be:

```
<span id="status"><i class="fa fa-circle text-orange"></i> DNS service on port </span>
```

without a port. 

I think you can add ` is_null($pistatus) ` to the condition on line 339.